httptools/parser/parser.pyx

```diff
@@ -117,7 +122,8 @@ cdef class HttpParser:
             self._current_header_value += val
 
     cdef _on_headers_complete(self):
-        self._maybe_call_on_header()
+        if self._current_header_name is not None:
```

Comment:
Not sure why you moved this check out of `_maybe_call_on_header` (so it's now duplicated).
Seems to me you could keep it in there, but just move the reset of `_current_XXX` out of if, something like:

```python
def _maybe_call_on_header():
   if self._current_header_name is not None:
       # do the real call
   # do the reset
```

Thoughts?